addons/mrp/models/mrp_production.py

```diff
@@ -794,6 +804,12 @@ def _compute_is_delayed(self):
                     record.date_deadline and (record.date_deadline < datetime.datetime.now() or record.date_deadline < record.date_finished))
             )
 
+    def _search_date_category(self, operator, value):
+        if operator != '=':
+            raise NotImplementedError(_('Operation not supported'))
+        search_domain = self.env['stock.picking'].date_category_to_domain(value)
+        return [('date_start', operator, value) for operator, value in search_domain]
```

Comment:
It's often good to use expression.AND in order to avoid unwanted problem in domain

Reply:
Done :)